src/Storage/StorageOperationTypes/StorageOperationTypes.php

```diff
@@ -0,0 +1,581 @@
+        return $this->found;
+    }
+
+    public function type(): ?string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @return float|int|string|null
+     */
+    public function value()
+    {
+        switch ($this->type) {
+            case StorageValueType::STRING:
+                return $this->value_string;
+            case StorageValueType::INTEGER:
+                return $this->value_int;
+            case StorageValueType::DOUBLE:
+                return $this->value_double;
+            case StorageValueType::BYTES:
+                return $this->value_bytes;
+            default:
+                return null;
+        }
+    }
+
+    public function valueString(): ?string
+    {
+        return $this->value_string;
+    }
+
+    public function valueInteger(): ?int
+    {
+        return $this->value_int;
+    }
+
+    public function valueDouble(): ?float
+    {
+        return $this->value_double;
+    }
+
+    public function valueBytes(): ?string
+    {
+        return $this->value_bytes;
+    }
+
+}
+
+/**
+ * Indicates that the request that generated it was successful.
+ */
+class StorageGetSuccess extends StorageGetResponse
+{
+    public function __construct(?_StoreGetResponse $grpcResponse=null)
+    {
+        parent::__construct();
+        if (!$grpcResponse) {
+            $this->found = false;
+            return;
+        }
+        $value = $grpcResponse->getValue();
+        if ($value->hasStringValue()) {
+            $this->type = StorageValueType::STRING;
+            $this->value_string = $value->getStringValue();
+        } elseif ($value->hasIntegerValue()) {
+            $this->type = StorageValueType::INTEGER;
+            $this->value_int = $value->getIntegerValue();
+        } elseif ($value->hasDoubleValue()) {
+            $this->type = StorageValueType::DOUBLE;
+            $this->value_double = $value->getDoubleValue();
+        } elseif ($value->hasBytesValue()) {
+            $this->type = StorageValueType::BYTES;
+            $this->value_bytes = $value->getBytesValue();
+        }
+    }
+
+    public function __toString()
+    {
+        $value = null;
+        switch ($this->type) {
+            case StorageValueType::STRING:
+                $value = $this->shortValue($this->value_string);
+                break;
+            case StorageValueType::INTEGER:
+                $value = $this->value_int;
+                break;
+            case StorageValueType::DOUBLE:
+                $value = $this->value_double;
+                break;
+            case StorageValueType::BYTES:
+                $value = $this->shortValue($this->value_bytes);
+                break;
+        }
+        return parent::__toString() . ": $value";
+    }
+}
+
+/**
+ * Contains information about an error returned from the request.
+ */
+class StorageGetError extends StorageGetResponse
+{
+    use ErrorBody;
+
+    public function type(): string
+    {
+        throw $this->innerException();
+    }
+
```

Comment:
Are we throwing exceptions here vs returning `null`?

Reply:
Good question, I wasn't sure how far to take this. Same question applies to `found()` as well, which makes sense as both `null` and `false` in the error class.

Reply:
I switched it so both `type()` and `found()` return `null` from the error class.